backend/src/main/java/org/track1/clog/controller/LogDetailController.java

```diff
@@ -54,21 +54,55 @@ public ResponseEntity<List<LogDetail>> getByGlobalInstanceId(@RequestParam(requi
     }
 
     //no longer accepts ranges for severity
-    public Specification<LogDetail> getByDatesPriority(Integer priority_low, Integer priority_high, Timestamp start, Timestamp end, Example<LogDetail> example){
+    public Specification<LogDetail> getByDatesPriority(boolean sev_info, boolean sev_warn, boolean sev_err, boolean priority_low, boolean priority_med, boolean priority_high, Timestamp start, Timestamp end, Example<LogDetail> example){
         return  (root,query,builder) -> {
             final List<Predicate> predicates = new ArrayList<Predicate>();
-            if (priority_low != null){
-                predicates.add(builder.greaterThanOrEqualTo(root.get("priority"), priority_low));
+            //predicates for each type of severity
+            Predicate predInfo = builder.between(root.get("severity"), 10, 29);
+            Predicate predWarn = builder.between(root.get("severity"),30,49);
+            Predicate predErr = builder.greaterThanOrEqualTo(root.get("severity"),50);
+            Predicate allSevPred = builder.disjunction();
+
+            //predicates for each type of severity
```

Comment:
Do you mean "predicates for each type of priority"?

Reply:
No, allSevPred is the combination of all severity predicates, allPrioPred is for priority